app/src/test/java/ch/epfl/favo/common/DatabaseWrapperTest.java

```diff
@@ -0,0 +1,143 @@
+    mockCollectionWrapper = new CollectionWrapper<>("favors", Favor.class);
+
+    // return collection refernece from firestore object
+    Mockito.doReturn(mockCollectionReference).when(mockFirestore).collection(anyString());
+    setupMockGetDocument();
+    setupMockDocumentListRetrieval();
+    DependencyFactory.setCurrentFirestore(mockFirestore);
+  }
+
+  /** Mock collectionreference->querysnapshotTask->querysnapshot */
+  private void setupMockDocumentListRetrieval() {
+    Task<QuerySnapshot> querySnapshotTask = Mockito.mock(Task.class);
+    Mockito.doReturn(querySnapshotTask).when(mockCollectionReference).get();
+    mockQuerySnapshot = Mockito.mock(QuerySnapshot.class);
+
+    Mockito.doReturn(mockQuerySnapshot).when(querySnapshotTask).getResult();
+    Mockito.doReturn(mockQuerySnapshot).when(querySnapshotTask).getResult(any());
+  }
+
+  /** Mock DocumentReference -> documentSnapshotTask->documentsnapshot */
+  private void setupMockGetDocument() {
+    // return document reference when collection.document()
+    Mockito.doReturn(mockDocumentReference).when(mockCollectionReference).document(anyString());
+    Mockito.doReturn(mockDocumentReference).when(mockCollectionReference).document();
+    // mock return task
+    documentSnapshotTask = Mockito.mock(Task.class);
+    // mock document returned by task
+    mockDocumentSnapshot = Mockito.mock(DocumentSnapshot.class);
+    // return mock document when task.getresult
+    Mockito.doReturn(mockDocumentSnapshot).when(documentSnapshotTask).getResult();
+    // return task when document reference is called on get
+    Mockito.doReturn(documentSnapshotTask).when(mockDocumentReference).get(any());
+    Mockito.doReturn(documentSnapshotTask).when(mockDocumentReference).get();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    DependencyFactory.setCurrentCompletableFuture(null);
+    DependencyFactory.setCurrentFirestore(null);
+  }
+
+  @Test
+  public void addDocument() {
+
+    mockCollectionWrapper.addDocument(testFavor);
+  }
+
+  @Test
+  public void removeDocument() {
+    mockCollectionWrapper.removeDocument("bla");
+  }
+
+  @Test
+  public void updateDocument() {
+    mockCollectionWrapper.updateDocument("bu", testFavor.toMap());
+  }
+
+  @Test
+  public void testGetDocumentReturnsExpectedDocument()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    // return favor object from document
+    Mockito.doReturn(testFavor).when(mockDocumentSnapshot).toObject(any());
+    Mockito.doReturn(true).when(mockDocumentSnapshot).exists();
+    CompletableFuture<DocumentSnapshot> futureSnapshot = new CompletableFuture<>();
+    futureSnapshot.complete(mockDocumentSnapshot);
+    DependencyFactory.setCurrentCompletableFuture(futureSnapshot);
+    CompletableFuture<Favor> actualFuture = mockCollectionWrapper.getDocument("fish");
+    Favor obtained = actualFuture.get(2, TimeUnit.SECONDS);
+    assert (obtained.equals(testFavor));

```

Comment:
Why using assert instead of junit.Assert?

Reply:
Updated